support-frontend/app/views/MainTypes.scala

```diff
@@ -0,0 +1,15 @@
+package views
+
+import play.twirl.api.Html
+
+sealed trait ReactDiv {
+  def fold[C](fa: EmptyDiv => C, fb: SSRContent => C): C = (this match {
+    case e: EmptyDiv => Left(e)
+    case r: SSRContent => Right(r)
+  }).fold(fa, fb)
```

Comment:
if we make our own sealed trait in the vision of Either, it doesn't get a fold method.  So I have just written a fold that implements toEither and then uses Either's fold as that seemed neatest.